lib/Sema/ConstraintSystem.cpp

```diff
@@ -7311,24 +7311,13 @@ bool ConstraintSystem::participatesInInference(ClosureExpr *closure) const {
   if (getAppliedResultBuilderTransform(closure))
     return true;
 
-  if (closure->hasSingleExpressionBody())
-    return true;
-
   if (Options.contains(ConstraintSystemFlags::LeaveClosureBodyUnchecked))
     return false;
```

Comment:
Ah, I think if we're keeping the `LeaveClosureBodyUnchecked` flag, then we'll probably want to restore the `hasSingleExpressionBody` check since I assume `LeaveClosureBodyUnchecked` is only meant to apply to multi-statement closures?

Reply:
You might be right. But things are looking good right now even without the check. Let’s see what the stress tester has to say

Reply:
Appears to be fine even with `closure->hasSingleExpressionBody` removed.

Reply:
I don't think `hasSingleExpressionBody` is necessary and you can remove `isInResultBuilderContext` method too since this was the last use of it. The only question I have is why do we have this setup in a way that `getAppliedResultBuilderTransform` doesn't respect `LeaveClosureBodyUnchecked`...

I think this method should look as follows:

```c++
return !(Options.contains(ConstraintSystemFlags::LeaveClosureBodyUnchecked) || closure->hasEmptyBody());
```

Reply:
Good point, I think we should add an method to get a body for a closure and use the result to check whether it's empty or not, seems like that would minimize the risk of making mistakes like that...